spotify-artist-viewer/Networking/SARequestManager.m

```diff
@@ -2,22 +2,112 @@
 //  SARequestManager.m
 //  spotify-artist-viewer
 //
+//  Created by Sara Lieto on 6/9/15.
 //  Copyright (c) 2015 Intrepid. All rights reserved.
 //
 
 #import "SARequestManager.h"
+#import "SAArtist.h"
 
 @implementation SARequestManager
 
 + (instancetype)sharedManager {
-    // TODO: return a singleton instance of SARequestManager
-    return nil;
+    static dispatch_once_t onceToken;
+    static SARequestManager *sharedManager = nil;
+    dispatch_once(&onceToken, ^{
+        sharedManager = [SARequestManager new];
+    });
+    return sharedManager;
 }
 
-- (void)getArtistsWithQuery:(NSString *)query
-                    success:(void (^)(NSArray *artists))success
-                    failure:(void (^)(NSError *error))failure {
-    // TODO: make network calls to spotify API
+- (void) getArtistWithQuery:(NSString *) query success:(void (^)(NSArray *artists))success failure:(void(^)(NSError *error))failure{
+    //Construction of URL with current Search Bar Text (query)
+    NSString *urltest = @"https://api.spotify.com/v1/search?q=";
+    NSString *url = [urltest stringByAppendingString:query];
+    NSString *type = @"&type=artist";
+    NSString *urlWithType = [url stringByAppendingString:type];
+    NSURL *requestURL = [NSURL URLWithString:urlWithType];
```

Comment:
These lines for building a request url would be a good place to refactor your code. You could have a method called -(NSURL *)configureRequestURLForQueryTerm:(NSString *)term. This will move the url creation out of this method to make the code cleaner and more maintainable.